<?xml version="1.0"?>
<launch>
    <arg name="steps_per_phase" default="10"/>
    <arg name="plotting" default="true" />
    <arg name="file_write_cb" default="false" />
    <arg name="box_side" default="0.2"/>

    <arg name="box_x" default="-0.3"/>
    <arg name="box_y" default="1.55"/>
    <arg name="box_z" default="1.45"/>

    <arg name="box_parent_link" default="world"/>
    <arg name="rviz" default="true"/>
    <arg name="testing" default="false"/>

    <!-- Load Robot Model    -->
    <param name="robot_description"  command="$(find xacro)/xacro --inorder $(find pick_and_place_support)/urdf/robot.urdf.xacro"/>

    <!-- Load Robot Semantic -->
    <param name="robot_description_semantic" textfile="$(find pick_and_place_support)/urdf/carbody_ur10_robot.srdf"/>

    <!-- Set simulated box parameters -->
    <param name="box_side" value="$(arg box_side)"/>
    <param name="box_z" value= "$(arg box_z)"/>
    <param name="box_x" value="$(arg box_x)"/>
    <param name="box_y" value="$(arg box_y)"/>

    <param name="box_parent_link" value="$(arg box_parent_link)"/>
    <group unless="$(arg testing)">

      <!-- Launch GUI to drive joints   -->
      <node name="joint_state_publisher" pkg="joint_state_publisher" type="joint_state_publisher" />

      <!-- Launch robot state publisher -->
      <node name="robot_state_publisher" pkg="robot_state_publisher" type="robot_state_publisher" />

      <!-- Launch planner node -->
      <node name="pick_and_place_node" pkg="pick_and_place" type="pick_and_place_node" output="screen">
          <param name="steps_per_phase" value="$(arg steps_per_phase)"/>
          <param name="plotting" value="$(arg plotting)"/>
          <param name="file_write_cb" value="$(arg file_write_cb)" />
          <param name="rviz" type="bool" value="$(arg rviz)"/>
      </node>

      <!-- Launch visualization -->
      <node if="$(arg rviz)" pkg="rviz" type="rviz" name="unicar_demo_pick_and_place_example_rviz"
          args="-d $(find pick_and_place_support)/config/examples.rviz" />
    </group>

</launch>
